pkg/orchestrator/scanwatcher/helpers.go

```diff
@@ -67,57 +67,60 @@ func newAssetScanFromScan(scan *models.Scan, assetID string) (*models.AssetScan,
 		ScanFamiliesConfig:            familiesConfig,
 		ScannerInstanceCreationConfig: scan.AssetScanTemplate.ScannerInstanceCreationConfig,
 		Status: &models.AssetScanStatus{
-			Exploits: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Exploits),
-			},
 			General: &models.AssetScanState{
 				Errors: nil,
 				State:  utils.PointerTo(models.AssetScanStateStatePending),
 			},
-			Malware: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Malware),
-			},
-			Misconfigurations: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Misconfigurations),
-			},
-			Rootkits: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Rootkits),
-			},
-			Sbom: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Sbom),
-			},
-			Secrets: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Secrets),
-			},
-			Vulnerabilities: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.Vulnerabilities),
-			},
-			InfoFinder: &models.AssetScanState{
-				Errors: nil,
-				State:  getInitStateFromFamilyConfig(familiesConfig.InfoFinder),
-			},
 		},
 		ResourceCleanupStatus: models.NewResourceCleanupStatus(
 			models.ResourceCleanupStatusStatePending,
 			models.ResourceCleanupStatusReasonAssetScanCreated,
 			nil,
 		),
+		Sbom: &models.SbomScan{
+			Packages: nil,
+			Status:   mapFamilyConfigToScannerStatus(familiesConfig.Sbom),
+		},
+		Exploits: &models.ExploitScan{
+			Exploits: nil,
+			Status:   mapFamilyConfigToScannerStatus(familiesConfig.Exploits),
+		},
+		Vulnerabilities: &models.VulnerabilityScan{
+			Vulnerabilities: nil,
+			Status:          mapFamilyConfigToScannerStatus(familiesConfig.Vulnerabilities),
+		},
+		Malware: &models.MalwareScan{
+			Malware:  nil,
+			Metadata: nil,
+			Status:   mapFamilyConfigToScannerStatus(familiesConfig.Malware),
+		},
+		Rootkits: &models.RootkitScan{
+			Rootkits: nil,
+			Status:   mapFamilyConfigToScannerStatus(familiesConfig.Rootkits),
+		},
+		Secrets: &models.SecretScan{
+			Secrets: nil,
+			Status:  mapFamilyConfigToScannerStatus(familiesConfig.Secrets),
+		},
+		Misconfigurations: &models.MisconfigurationScan{
+			Misconfigurations: nil,
+			Scanners:          nil,
+			Status:            mapFamilyConfigToScannerStatus(familiesConfig.Misconfigurations),
+		},
+		InfoFinder: &models.InfoFinderScan{
+			Infos:    nil,
+			Scanners: nil,
+			Status:   mapFamilyConfigToScannerStatus(familiesConfig.InfoFinder),
+		},
 	}, nil
 }
 
-func getInitStateFromFamilyConfig(config models.FamilyConfigEnabler) *models.AssetScanStateState {
+func mapFamilyConfigToScannerStatus(config models.FamilyConfigEnabler) *models.ScannerStatus {
```

Comment:
I like that the previous naming indicated that this function was retrieving the "initial" scanner status, not sure if the new naming reflects this. maybe consider changing the naming here?